core/extension/gdextension.cpp

```diff
@@ -716,6 +717,22 @@ Error GDExtension::open_library(const String &p_path, const String &p_entry_symb
 	}
 #endif
 
+	if (!p_lookup_paths.is_empty() && !FileAccess::exists(abs_path) && !DirAccess::exists(abs_path)) {
+		// Try using custom lookup paths.
+		for (const String &dir : p_lookup_paths) {
+			String lookup_dir = dir;
+			lookup_dir = lookup_dir.begins_with("pck_dir://") ? lookup_dir.replace_first("pck_dir:/", ProjectSettings::get_singleton()->get_main_pack_path()) : lookup_dir;
+			lookup_dir = lookup_dir.begins_with("bundle_dir://") ? lookup_dir.replace_first("bundle_dir:/", OS::get_singleton()->get_bundle_resource_dir()) : lookup_dir;
+			lookup_dir = lookup_dir.begins_with("exe_dir://") ? lookup_dir.replace_first("exe_dir:/", OS::get_singleton()->get_executable_path().get_base_dir()) : lookup_dir;
```

Comment:
It would be good to replace this with the custom path mapping from PR https://github.com/godotengine/godot/pull/87586 after that is merged, so that these paths aren't exclusive to GDExtension loading. Perhaps a TODO should be added here?

Also, are we sure we want to commit to these specific prefix names? Not that I want to spend lots of time bike-shedding the names :-) but we will be committing to them as part of our external API, so they probably should go through a little bit wider discussion.